tests/preserve_sequence.rs

```diff
@@ -0,0 +1,59 @@
+use ron::ser::{to_string_pretty, PrettyConfig};
+use ron::de::from_reader;
+use serde::Deserialize;
+use serde::Serialize;
+use std::{collections::HashMap, fs::File};
+use std::env;
+use std::fs;
+
+#[derive(Debug, Deserialize, Serialize)]
+struct Config {
+    boolean: bool,
+    float: f32,
+    map: HashMap<u8, char>,
+    nested: Nested,
+    tuple: (u32, u32),
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct Nested {
+    a: String,
+    b: char,
+}
+
+fn read_original(file_name: &str) -> (String) {
+    let input_path = format!("{}/tests/{}", env!("CARGO_MANIFEST_DIR"), file_name);
```

Comment:
Why don't we just include the original with `include_str!` or something?

Reply:
Didn't know it exists. However, if I try, the compiler keeps complaining "error: argument must be a string literal", unless I copy the string into the function. But then the flexibility of calling various files is gone.

Reply:
What I was thinking:
```rust
fn read_original(source: &str) -> String {...}

fn test_xxx() {
  let file1 = include_str!("my_file1.ron");
  assert_eq!(
    read_original(file1),
    make_roundtrip(file1),
  );  
}
```

Reply:
Ah, okay. Then the file is only read once, and processed twice. A much nicer way indeed. See 0fa9da321841aa6dea640007007348000a67f9a0